bookshelf.js

```diff
@@ -0,0 +1,76 @@
+window.bookshelf = (function() {
+	var buttonNode = document.createElement("button");
+	buttonNode.setAttribute("id", 'fetch-books-btn');
+	var textNode = document.createTextNode("Fetch Books");
+	buttonNode.appendChild(textNode);
+	buttonNode.addEventListener("click", fetchBooks);
+	var body = document.getElementsByTagName("body")[0];
+	body.appendChild(buttonNode);
+
+	//function createBtn() {
+		function fetchBooks() {
+			const booksJSON = 'https://raw.githubusercontent.com/codeyourfuture/bookshelf-project/master/books.json'
+			fetch(booksJSON).then(response => response.json()).then(processBooks => {
+				var x = document.createElement("UL");
+				x.setAttribute("id", "myUL");
+				document.body.appendChild(x);
+				processBooks.forEach(function(xz) {
+					var y = document.createElement("LI");
+					y.setAttribute("id", xz.id);
+					var b = document.createElement("button");
+					var bu = document.createTextNode("⬆");
+					b.setAttribute("class", 'moveUp');
+					b.appendChild(bu);
+					 b.addEventListener("click", moveUp);
+					var but = document.createElement("button");
+					var butto = document.createTextNode("⬇");
+					but.setAttribute("class", 'moveDown');
+					but.appendChild(butto);
+					but.addEventListener("click",moveDown );
+					var t = document.createTextNode(` -  ${xz.title} by ${xz.author}`);
+					y.appendChild(b);
+					y.appendChild(but);
+					y.appendChild(t);        
+                    x.appendChild(y);
+                    
+					x.addEventListener('click', function(e) {
+						var clicked = e.target;
+						var li = clicked.parentNode;
+						//var next = li.nextElementSibling;
+						//if (next != null) next = next.nextElementSibling;
+						//var prev = li.previousElementSibling;
+						// if (prev != null) prev = prev.previousElementSibling;
+						if (clicked.className === 'moveDown') {
+							x.insertBefore(li, xz.id.nextSibling);
```

Comment:
pay attention on how you use the `insertBefore`, gave a look to the [documentation](https://developer.mozilla.org/en-US/docs/Web/API/Node/insertBefore). Remember that the prameters have to be 2 nodes, the first is the new node you want to insert, while the second is the reference node.